xamarin/Xamarin.Forms
# [WPF] Open DatePicker when control receives focus (to match other platforms)

### Description of Change ###

Fixes consistency with other platforms. When the control receives keyboard focus open the dropdown.

### Issues Resolved ### 
- fixes #10358

### API Changes ###
 
 None

### Platforms Affected ### 

- WPF

### Behavioral/Visual Changes ###
Previously the control would only open when user tapped on the calendar button. Now the control opens automatically when you call Focus() on it programmatically.

### Before/After Screenshots ### 
Not applicable

### Testing Procedure ###
Use the control gallery app
Go to DatePicker Gallery
Select Focus test
Press the button to set focus to the DatePicker and observe the drop-down calendar is displayed

### PR Checklist ###
